submissions/zalevskyi/frogger-game/app.js

```diff
@@ -0,0 +1,340 @@
+    constructor(center, radius) {
+        this.center = center
+        this.radius = radius
+    }
+    //checks overlap between two instances of Circle
+    static isOverlap(c1, c2) {
+        return c1.radius + c2.radius > Point2D.distance(c1.center, c2.center)
+    }
+}
+/***Game Entities***/
+//Basic sprite class to render, change position and check collisions
+class Sprite {
+    constructor(image, position) {
+        // image - url of the sprite image
+        // position - Point2D instance of upper-left image corner
+        this.image = image
+        this.setInitialPosition()
+        this.allPoints = [this.position, ...this.collisionCircles.map(circle => circle.center)]
+        this.changePosition(position.x - this.position.x, position.y - this.position.y)
+    }
+    setInitialPosition() {
+        this.position = new Point2D(0, 0)
+        this.collisionCircles = collisionCirclesFactory(this.image)
+    }
+    render() {
+        ctx.drawImage(Resources.get(this.image), this.position.x, this.position.y);
+    }
+    //changes absolute position of sprite image and all its collision circles
+    changePosition(delta_x, delta_y) {
+        this.allPoints.forEach(point => {
+            point.x += delta_x
+            point.y += delta_y
+        })
+    }
+    //checks collision between two sprites
+    static isCollision(s1, s2) {
+        return s1.collisionCircles.reduce((result, s1_circle) =>
+                result || s2.collisionCircles.reduce((result, s2_circle) =>
+                    result || Circle.isOverlap(s1_circle, s2_circle), false), false
+            )
+    }
+}
+class Collectable extends Sprite {
+    constructor(image, position) {
+        super(image, position)
+        this.timeToAppear = false
+        this.timeToDisappear = false
+    }
+    render() {
+        if (this.timeToAppear && this.timeToAppear<Date.now()) this.appear()
+        if (this.timeToDisappear && this.timeToDisappear<Date.now()) this.disappear()
+        super.render()
+    }
+    appear() {
+        this.changePosition(Math.random()*404 - this.position.x, 0)
+        this.timeToAppear = false
+        this.timeToDisappear = Date.now() + 1000 * 6
+    }
+    disappear() {
+        this.changePosition(OFF_SCREEN_X - this.position.x, 0)
+        this.timeToAppear = Date.now() + 1000 * 3
+        this.timeToDisappear = false
+    }
+}
+//Sprite that can move on its own
+class Actor extends Sprite {
+    constructor(image, position, speed) {
+        super(image, position)
+        this.speed = speed
+        this.moveStatus = {'down': false, 'up': false, 'right': false, 'left': false}
+    }
+    // Parameter: dt, a time delta between ticks
+    // You should multiply any movement by the dt parameter
+    // which will ensure the game runs at the same speed for all computers
+    update(dt) {
+        let move_x = this.moveStatus['right']-this.moveStatus['left']
+        let move_y = this.moveStatus['down']-this.moveStatus['up']
```

Comment:
JS is a loosely typed language and offers rich value coercion rules. This is hardly a benefit, otherwise the community wouldn't require TypeScript to fix the situation.

Anyway attempt to perform purely arithmetic operations over potentially boolean values is not only confusing but is also a somewaht error prone approach.

Think of a peer developer that would have to expand or change your code. 

Looks like an artifact from a different approach.